File: airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/credential/GcsHmacKeyCredentialConfig.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.gcs.credential;

import com.fasterxml.jackson.databind.JsonNode;
import io.airbyte.integrations.destination.s3.credential.S3AccessKeyCredentialConfig;
import io.airbyte.integrations.destination.s3.credential.S3CredentialConfig;
import java.util.Optional;

public class GcsHmacKeyCredentialConfig implements GcsCredentialConfig {

  private final String hmacKeyAccessId;
  private final String hmacKeySecret;

  public GcsHmacKeyCredentialConfig(final JsonNode credentialConfig) {
    this.hmacKeyAccessId = credentialConfig.get("hmac_key_access_id").asText();
    this.hmacKeySecret = credentialConfig.get("hmac_key_secret").asText();
  }

  public GcsHmacKeyCredentialConfig(final String hmacKeyAccessId, final String hmacKeySecret) {
    this.hmacKeyAccessId = hmacKeyAccessId;
    this.hmacKeySecret = hmacKeySecret;
  }

  public String getHmacKeyAccessId() {
    return hmacKeyAccessId;
  }

  public String getHmacKeySecret() {
    return hmacKeySecret;
  }

  @Override
  public GcsCredentialType getCredentialType() {
    return GcsCredentialType.HMAC_KEY;
  }

  @Override
  public Optional<S3CredentialConfig> getS3CredentialConfig() {
    return Optional.of(new S3AccessKeyCredentialConfig(hmacKeyAccessId, hmacKeySecret));
  }

}

File: airbyte-cdk/java/airbyte-cdk/gcs-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/gcs/credential/GcsHmacKeyCredentialConfig.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.gcs.credential

import com.fasterxml.jackson.databind.JsonNode
import io.airbyte.cdk.integrations.destination.s3.credential.S3AccessKeyCredentialConfig
import io.airbyte.cdk.integrations.destination.s3.credential.S3CredentialConfig
import java.util.*

class GcsHmacKeyCredentialConfig : GcsCredentialConfig {
    val hmacKeyAccessId: String
    val hmacKeySecret: String

    constructor(credentialConfig: JsonNode) {
        this.hmacKeyAccessId = credentialConfig["hmac_key_access_id"].asText()
        this.hmacKeySecret = credentialConfig["hmac_key_secret"].asText()
    }

    constructor(hmacKeyAccessId: String, hmacKeySecret: String) {
        this.hmacKeyAccessId = hmacKeyAccessId
        this.hmacKeySecret = hmacKeySecret
    }

    override val credentialType: GcsCredentialType
        get() = GcsCredentialType.HMAC_KEY

    override val s3CredentialConfig: Optional<S3CredentialConfig>
        get() = Optional.of(S3AccessKeyCredentialConfig(hmacKeyAccessId, hmacKeySecret))
}
